### PR TITLE
Document and cleanup Wallet.py

### PR DESCRIPTION
**Pull request by ixje**
**What current issue(s) does this address?, or what feature is it adding?**
Contributes to addressing #63 

**How did you solve this problem?**
write write write

**How did you make sure your solution works?**
-

**Did you add any tests?**
-

**Did you check your `pycodestyle`?**
yes

**Are there any special changes in the code that we should be aware of?**
just a small update to UserWallet.CreateKey / Wallet.Createkey()  nothing that breaks others calling it

**Are you making a PR to a feature branch or development rather than master?**
development

### CODE REVIEW

**Comment by ixje** [comment]
@localhuman I believe you made most commits in the past in this file, so I think you'd be best suited to check if the docsstrings make sense/are correct. I tried to verify the arguments and return types as much as possible and add a sensible description.

**Comment by localhuman** [comment]
Looks like the test build failed.  Also please refer to this comment for doc string format https://github.com/CityOfZion/neo-python/pull/73#issuecomment-342985048

**Comment by metachris** [comment]
fyi there is no `void` in Python, methods return `None` by default.

**Comment by ixje** [comment]
Thanks for pointing out. I wasn't sure what to do with that and you guys were all asleep / at work when I was writing. I have to update it to the rST format anyway, so I'll fix that as well.

**Comment by ixje** [comment]
@localhuman as discussed

Functions that require a description:
FindCoinsByVins
FindUnspentCoins
FindUnspentCoinsByAsset
FindUnspentCoinsByAssetAndTotal
ProcessBlocks
ProcessNewBlock
Rebuild
MakeTransaction

Functions that have an argument/return type I'm not 100% sure about (see the TODO in the docstring):
DeleteAddress
Sign
SaveTransaction

File: neo/Wallets/Wallet.py [hunk 1]
```diff
@@ -122,38 +121,76 @@ def __init__(self, path, passwordKey, create):
...
     def AddWatchOnly(self, script_hash):
+        """
+        Add a watch only address to the wallet.
 
+        Args:
+            script_hash (UInt160): a bytearray (len 20) representing the public key.
+
+        Note:
+            Prints a warning to the console if the address already exists in the wallet.
+
```
Comment (metachris) [inline neo/Wallets/Wallet.py]: Could you please remove the empty line at the end of each docstring?

File: neo/Wallets/Wallet.py [hunk 2]
```diff
@@ -165,42 +202,124 @@ def ChangePassword(self, password_old, password_new):
...
+        Determine if the wallet contains the address.
+
+        Args:
+            address (str): a string representing the public key.
+
+        Returns:
+            bool: True, if the address is present in the wallet. False otherwise.
+
+        """
+
```
Comment (metachris) [inline neo/Wallets/Wallet.py]: Also I'd say don't add empty lines after the docstring, start with the code straight away.

File: neo/Implementations/Wallets/peewee/UserWallet.py [hunk 3]
```diff
@@ -87,21 +84,14 @@ def Create(path, password):
         return wallet
 
     def CreateKey(self, prikey=None):
-        if prikey:
-            private_key = prikey
-        else:
-            private_key = bytes(Random.get_random_bytes(32))
-
-        account = WalletKeyPair(priv_key=private_key)
+        account = super(UserWallet, self).CreateKey(private_key=prikey)
```
Comment (metachris) [inline neo/Implementations/Wallets/peewee/UserWallet.py]: I'm not 100% sure about this. What if the `Wallet.CreateKey` method changes to do something more? You could just simplify the block on top with `private_key = prikey or bytes(Random.get_random_bytes(32))`

**Comment by metachris** [comment]
Good job, a lot of initial much needed docstrings! I've just got a few minor suggestions (see above): (no newline at end of docstring and after docstring, and one possible code issue). 👍

File: neo/Implementations/Wallets/peewee/UserWallet.py (hunk 3 above)
Comment (ixje) [reply neo/Implementations/Wallets/peewee/UserWallet.py]: my changes actually make it match the original `neo-project` more closely
See: https://github.com/neo-project/neo/blob/1bbebb1db8fa7e5a72f372e933f65f4815a1911b/neo/Implementations/Wallets/EntityFramework/UserWallet.cs#L102

Comment (metachris) [reply neo/Implementations/Wallets/peewee/UserWallet.py]: Thanks, good to have this context!

**Comment by localhuman** [comment]
Great work!  Will work on filling in what is missing-

**Pull request merged**